Bluejay/Bluejay/Bluejay.swift

```diff
@@ -129,24 +132,18 @@ public class Bluejay: NSObject {
     }
     
     /**
-     Starting Bluejay will initialize the CoreBluetooth stack. Initializing a Bluejay instance will not yet initialize the CoreBluetooth stack. An explicit call to start running a Bluejay instance after it is intialized is required because in cases where a state resotration is trying to restore a listen on a characteristic, a listen restorer must be available before the CoreBluetooth stack is re-initialized. This two-step startup allows you to insert and gaurantee the setup of your listen restorer in between the initialization of Bluejay and the initialization of the CoreBluetooth stack.
+     Starting Bluejay will initialize the CoreBluetooth stack. Simply initializing a Bluejay instance without calling this function will not initialize the CoreBluetooth stack. An explicit start call is required because in cases where a state resotration is trying to restore a listen on a characteristic, a listen restorer must be available before the CoreBluetooth stack is re-initialized. This two-step startup (init then start) allows you to insert and gaurantee the setup of your listen restorer in between the initialization of Bluejay and the initialization of the CoreBluetooth stack.
      
      - Parameters:
+        - mode: CoreBluetooth initialization modes and options.
         - observer: An object interested in observing Bluetooth connection events and state changes. You can register more observers using the `register` function.
-        - restoreMode: Determines whether Bluejay will opt-in to state restoration, and if so, can optionally provide a listen restorer as well for restoring listens.
-        - coreBluetoothState: Allows starting Bluejay with an existing Core Bluetooth manager and peripheral.
     */
-    public func start(
-        connectionObserver observer: ConnectionObserver? = nil,
-        backgroundRestore restoreMode: BackgroundRestoreMode = .disable,
-        coreBluetoothState: (manager: CBCentralManager, peripheral: CBPeripheral?)? = nil
-        )
-    {
+    public func start(mode: StartMode, connectionObserver observer: ConnectionObserver? = nil) {
```

Comment:
Should `mode` maybe have a default of `.new(StartOptions.default)`, it's a little unfortunate to HAVE to pass in a start options in the case where just start() would have worked before. 

If it's not possible to do a default parameter pointing at another var like that, maybe there should at least be an alternate version of `start` that just takes a connection observer and calls the main start with the default.